src/ai/backend/manager/api/service.py

```diff
@@ -478,6 +480,20 @@ class NewServiceRequestModel(LegacyBaseRequestModel):
     )
     config: ServiceConfigModel
 
+    @field_validator("service_name")
+    @classmethod
+    def validate_service_name(cls, name: str) -> str:
+        service_name_matcher = re.compile(
+            r"^(?=.{4,%d}$)\w[\w-]*\w$" % (tv.SESSION_NAME_MAX_LENGTH,), re.ASCII
+        )
+        if not service_name_matcher.search(name):
+            raise InvalidAPIParameters(
+                f"Invalid service name format."
+                f"Service name must be 4-{tv.SESSION_NAME_MAX_LENGTH} characters long, and can only contain alphanumeric characters, underscores, and hyphens."
+                f"It must start and end with an alphanumeric character."
+            )
+        return name
```

Comment:
Let's check the length limit with the Field `ge` and make it only check the pattern so we can simply see the pattern matching.
In that case, it seems unnecessary to define a field_validator since it is already possible with just the pattern and ge in Field.